cpp/src/parquet/statistics.cc

```diff
@@ -610,8 +629,12 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     if (HasNullCount()) {
       s.set_null_count(this->null_count());
     }
-    // num_values_ is reliable and it means number of non-null values.
-    s.all_null_value = num_values_ == 0;
+    if (HasNullCount()) {
```

Comment:
This refactor is a bit tricky. When `ColumnChunkMetadata` read a statistics from thrift, and if the thrift not have `null_count`, the `num_values` would be wrong. And in C++ Parquet, we'll never meet this case, because writer always has `has_null_count_`

Reply:
Why would `num_values` be wrong?

Reply:
Can a `!metadata.statistics.__isset.null_count` might means:

1. The underlying data has null
2. But it doesn't write it into statistics

If it can, when `!metadata.statistics.__isset.null_count`, `metadata.num_values - metadata.statistics.null_count` cannot represent the non-null count